ngraph/src/ngraph/pass/nop_elimination.cpp

```diff
@@ -343,8 +341,9 @@ static bool eliminate_squeeze(const std::shared_ptr<Node>& node)
         return false;
     };
     // eliminate redundant unsqueeze->squeeze
-    if (unsqueeze)
+    if (auto unsqueeze = as_type_ptr<opset3::Unsqueeze>(input))
```

Comment:
I think you need to rebase your branch - as far as I'm concerned fixes from @GlebKazantaev for NopElimintaion have been merged already..